sphinx-search-as-you-type.rst

```diff
@@ -0,0 +1,113 @@
+are looking for has always been important for Read the Docs.
+This year, I, `Vaibhav Gupta`_, took the opportunity provided
+by Google Summer of Code to improve the search.
+The main goal of my `GSoC project`_ was to enable multiple hits per search result,
+improve the UI/UX around code search and to make an `Sphinx extension`_ to provide
+"search as you type" experience to the users.
+
+Google Summer of Code
+---------------------
+
+Google Summer of Code is a global program where students work with
+an open source organization on a 3-month programming project.
+I got accepted in the program and was excited to spend my summer with Read the Docs.
+
+I have worked full time during the GSoC coding period and implemented various features.
+All of my search related work can be seen in the `In-Doc Search UI Project Board`_.
+
+Background
+----------
+
+Search code was voluntarily contributed by `Rob Hudson`_,
+`back in 2013`_ and then improved by other contributors.
+It was greatly improved and upgraded by `Safwan Rahman`_ during :doc:`GSoC'18 <search-improvements>`.
+Continuining on the same path,
+I have implemented some new features on top of the existing search backend.
+
+New Features
+------------
+
+During the GSoC period, I have worked on the following features:
+
+- `In-Doc Search UI`_: We have worked on a Sphinx extension which adds a clean and minimal
+  search UI to your docs. It supports "search as you type" feature.
+  So, it is now possible to get instant results without being redirected to any other page.
+  Read the docs `here`_.
+
+.. image:: /_static/in-doc-search-demo.gif
+    :width: 100%
+    :target: /_static/in-doc-search-demo.gif
+
+- Multiple Hits Per Search Result: This is one of the highly requested features.
+  Now, we support results from the sections of the docs, clicking on which will take you
+  to that particular section and not just to the top of the result page.
+  You don't have to scroll anymore to the desired result.
+
+- Code Search: We now support code search. If you want to search a particular function
+  or an API endpoint -- you can just type your query and you will find it in the results.
+  Eg: ``api/v3/`` or ``module.function``.
+
+What Next?
+----------
+
+We don't intend to stop just yet.
+We are planning to work on some more cool features in the near future,
+some of which are:
+
+- **Search Facets**:
+  Facets can be used to make search more accurate.
+  Like ``subproject: project query``.
+- **Search Analytics**:
+  Analytics is always helpful to know what your users are actually looking for in your docs.
+- **Search Results Ordered By Most Viewed Pages**:
+  It would be much more useful if the most viewed pages are shown first in the search results.
+- **Search Inside Sections**:
+  It would be good if users have the option to get the
+  search results from a particular section of the documentation.
+  (`readthedocs/readthedocs-sphinx-search#23`_).
```

Comment:
It's great this has an issue attached. We should probably create issues for the other ideas mentioned in this post so people can follow along if they are interested. It would also help with contribution.